src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/events/handlers/PrisonerAppointmentsChangedEventHandler.kt

```diff
@@ -0,0 +1,70 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.handlers
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.prisonersearch.Prisoner
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.prisonersearch.PrisonerSearchClient
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.TimeSource
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.PrisonAppointmentRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.BookingFacade
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.UserService
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.PrisonerAppointmentsChangedEvent
+
+private const val RELEASED = "REL"
+private const val TRANSFERRED = "TRN"
+
+/**
+ * This event is raised on the back of a prison user deciding what to do with a prisoners appointments on the back of a
+ * movement (via NOMIS) e.g. transfer, release. They are given the option to cancel any existing appointments.
+ */
+@Component
+class PrisonerAppointmentsChangedEventHandler(
+  private val prisonAppointmentRepository: PrisonAppointmentRepository,
+  private val prisonerSearchClient: PrisonerSearchClient,
+  private val bookingFacade: BookingFacade,
+  private val timeSource: TimeSource,
+) : DomainEventHandler<PrisonerAppointmentsChangedEvent> {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  override fun handle(event: PrisonerAppointmentsChangedEvent) {
+    if (event.isCancelAppointments()) {
+      val now = timeSource.now()
+
+      val bookings = prisonAppointmentRepository.findActivePrisonerPrisonAppointmentsAfter(event.prisonerNumber(), now.toLocalDate(), now.toLocalTime())
+        .map { it.videoBooking.videoBookingId }
+        .distinct()
+
+      if (bookings.isNotEmpty()) {
+        val prisoner = prisonerSearchClient.getPrisoner(event.prisonerNumber()).throwNullPointerIfNotFound { "Unable to find prisoner ${event.prisonerNumber()}" }
+
+        when (prisoner.lastMovementTypeCode) {
+          RELEASED -> {
+            log.info("PRISONER_APPOINTMENTS_CHANGED_EVENT: processing release for prisoner ${event.prisonerNumber()} - $event")
```

Comment:
I may look at reducing the logging but for now would like include it just for monitoring.